### PR TITLE
Enabled custom validation for TypeConverters via Advanced.Validator

### PR DESCRIPTION
**Pull request by HenkKin**
If you use TypeConverters, this pull request allows you to add custom validators via Advanced.Validator

### CODE REVIEW

File: src/UnitTests/AdvancedConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,160 @@
+using System;
```
Comment (lbargaoanu) [inline src/UnitTests/AdvancedConfiguration.cs]: Can you please rename this file/class to CustomValidations? And move the existing test here.

Comment (HenkKin) [reply src/UnitTests/AdvancedConfiguration.cs]: I renamed it

Comment (lbargaoanu) [reply src/UnitTests/AdvancedConfiguration.cs]: https://github.com/AutoMapper/AutoMapper/blob/7a00700de61cd3234b6a6eb4bf0f6bbe402369b1/src/UnitTests/ConfigurationValidation.cs#L9 belongs in this file too.

Comment (HenkKin) [reply src/UnitTests/AdvancedConfiguration.cs]: This test is moved to the new CustomValidations file. I have searched for all usages of cfg.Advanced.Validator in test files, There is no one left.

**Pull request merged**